x-pack/legacy/plugins/graph/public/components/graph_search_bar.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFieldText,
+  EuiButton,
+  EuiPopover,
+  EuiButtonEmpty,
+  EuiToolTip,
+} from '@elastic/eui';
+import React, { useState } from 'react';
+
+import { CoreStart } from 'src/core/public';
+import { i18n } from '@kbn/i18n';
+import { IndexPatternSavedObject } from '../types/app_state';
+import { GraphSourcePicker } from './graph_source_picker';
+
+interface GraphSearchBarProps {
+  isLoading: boolean;
+  initialQuery?: string;
+  currentIndexPattern?: IndexPatternSavedObject;
+  onIndexPatternSelected: (indexPattern: IndexPatternSavedObject) => void;
+  onQuerySubmit: (query: string) => void;
+  savedObjects: CoreStart['savedObjects'];
+  uiSettings: CoreStart['uiSettings'];
+}
+
+export function GraphSearchBar({
+  currentIndexPattern,
+  onQuerySubmit,
+  isLoading,
+  onIndexPatternSelected,
+  initialQuery,
+  ...sourcePickerProps
+}: GraphSearchBarProps) {
+  const [open, setOpen] = useState(false);
+  const [query, setQuery] = useState(initialQuery || '');
+  return (
+    <form
+      className="gphSearchBar"
+      onSubmit={e => {
+        e.preventDefault();
+        if (!isLoading && currentIndexPattern) {
+          onQuerySubmit(query);
+        }
+      }}
+    >
+      <EuiFlexGroup gutterSize="m">
+        <EuiFlexItem>
+          <EuiFieldText
+            fullWidth
+            isLoading={isLoading}
+            icon="search"
+            placeholder={i18n.translate('xpack.graph.bar.searchFieldPlaceholder', {
+              defaultMessage: 'Search your data and add to your graph',
+            })}
+            prepend={
+              <EuiPopover
+                id="graphSourcePicker"
+                anchorPosition="downLeft"
+                ownFocus
+                button={
+                  <EuiToolTip
```

Comment:
There's a bit of a bug when using popovers and tooltips in prepends. I have a fix in for EUI, but in the mean time:

Add an `anchorClassName = "gphSearchBar__datasourceButtonTooltipAnchor"` and then remove the padding applied to that via:

```scss
.gphSearchBar__datasourceButtonTooltipAnchor {
  padding: 0;
}
```

<img src="https://d.pr/free/i/5oIBgJ+" width="50%" />

Reply:
The **New** click delay might be a regression #45701